### ISSUE TITLE
Docker image releases with different Go version than binaries

### ISSUE THREAD

**Issue by jdolitsky** [oras-project/oras#251]
```
$ oras version
Version:        0.11.1
Go version:     go1.16.2
Git commit:     5bfe0ab04a0f6fcdc5de4294cc79fb5b2e23a9a5
Git tree state: clean
```

```
$ docker run -it --rm -v $(pwd):/workspace ghcr.io/deislabs/oras:v0.11.1 version
Version:        0.11.1
Go version:     go1.15.7
Git commit:     5bfe0ab04a0f6fcdc5de4294cc79fb5b2e23a9a5
Git tree state: clean
```

How should we keep these best in sync other than using `latest` tag? Should we have a `GO_VERSION` file that is used both for CI and Dockerfile?

**Comment by deitch** [comment]
A few possibilities:

* set the go version in the `Makefile` and pass it via a `--build-arg` when building in Docker, via `ARG` in the `Dockerfile`
* build the distributed artifact using the `Dockerfile`
* do it manually 😄

**Comment by sajayantony** [comment]
Glad we don't have a latest. This makes pre-release pipelines easier since the tag would be the full version. For e.g. `v0.2.0-alpha.1`

**Comment by shizhMSFT** [comment]
Once we resolves #377, we will have the same go version for the same oras version.
